lib/ClangImporter/ImportName.cpp

```diff
@@ -576,12 +574,96 @@ determineCtorInitializerKind(const clang::ObjCMethodDecl *method) {
   return None;
 }
 
+template <typename A>
+static bool matchesVersion(A *versionedAttr, ImportNameVersion version) {
+  clang::VersionTuple attrVersion = versionedAttr->getVersion();
+  if (attrVersion.empty()) {
+    // FIXME: This is brittle. The empty version represents "newest", as in
+    // "newer than the current version", but ImportNameVersion::Swift4 is
+    // a specific new version.
+    return version == ImportNameVersion::Swift4;
```

Comment:
I considered having a "Latest" version in ImportNameVersion. You could go with that, or at the very least a global in the ImportName header that is set to Swift4